src/struct_mv/_hypre_struct_mv.h

```diff
@@ -2282,7 +2348,16 @@ HYPRE_Int hypre_ReadBoxArrayData ( FILE *file , hypre_BoxArray *box_array , hypr
 HYPRE_Int hypre_ReadBoxArrayData_CC ( FILE *file , hypre_BoxArray *box_array , hypre_BoxArray *data_space , HYPRE_Int stencil_size , HYPRE_Int real_stencil_size , HYPRE_Int constant_coefficient , HYPRE_Int dim , HYPRE_Complex *data );
 
 /* struct_matmult.c */
-HYPRE_Int hypre_StructMatmult ( HYPRE_Int nmatrices , hypre_StructMatrix **matrices , HYPRE_Int nterms , HYPRE_Int *terms , HYPRE_Int *transposes , hypre_StructMatrix **M_ptr );
+hypre_StructMMData* hypre_StructMatrixMultCreate ( HYPRE_Int nmatrices_in , hypre_StructMatrix **matrices_in , HYPRE_Int nterms , HYPRE_Int *terms_in , HYPRE_Int *transposes_in );
+HYPRE_Int hypre_StructMatrixMultDestroy ( hypre_StructMMData *mmdata );
+HYPRE_Int hypre_StructMatrixMultSetup ( hypre_StructMMData  *mmdata , hypre_StructMatrix **M_ptr );
+HYPRE_Int hypre_StructMatrixMultCommunicate ( hypre_StructMMData *mmdata );
```

Comment:
On second thought, maybe we should go back to `StructMatmult` without `Matrix`, at least for the internal routines, just for brevity.  The interfaces we expose to users can be longer.